crates/gitbutler-core/src/virtual_branches/base.rs

```diff
@@ -31,7 +34,7 @@ pub struct BaseBranch {
     pub behind: usize,
     pub upstream_commits: Vec<RemoteCommit>,
     pub recent_commits: Vec<RemoteCommit>,
-    pub last_fetched_ms: Option<u128>,
+    pub last_fetched_at: Option<Duration>,
```

Comment:
Again, I don't think that a `Duration` is the right abstraction for a timestamp, even though I *feel* that it's better than a u128. Ideally it could just be a timestamp, like `std::time::Instant` or `std::time::SystemTime` (all of which are UTC which seems to be all that's needed here). Then, when translating it for use in the frontend, the necessary conversions would have to be done.